lib/fetch.js

```diff
@@ -31,23 +67,40 @@ function buildRequestOptions(request) {
   }
 }
 
+/**
+ * Makes an HTTP request using axios with the specified options.
+ *
+ * @param {FetchRequestOptions} request - The request configuration options
+ * @param {AxiosStatic | AxiosInstance} [axiosInstance=axios] - Optional axios instance to use for the request. Default
+ *   is `axios`
+ * @returns {Promise<any | FetchResponse>} Promise that resolves to the response data or full response object
+ * @throws {FetchError} When the request fails or returns an error status code
+ */
 async function callFetch(request, axiosInstance = axios) {
   try {
     const options = buildRequestOptions(request)
     // @ts-ignore
     const response = await axiosInstance(options)
     if (!request.resolveWithFullResponse) return response.data
+    // @ts-ignore - Adding custom properties to response object
```

Comment:
Done to avoid

```
Property 'statusCode' does not exist on type 'AxiosResponse<any, any>'.ts(2339)
```

Reply:
As a future separate PR, we can potentially adapt all callers to use the equivalent fields in AxiosResponse to avoid adding custom properties to the response object

Reply:
Yeah, I agree. My approach at the moment is to avoid touching the underlying JavaScript code as much as is possible. Changing underlying code can come in a follow-up wave.